prerendering.bs

```diff
@@ -426,6 +431,20 @@ Patch the [=navigate=] algorithm to allow the [=prerendering browsing context/ac
     1. Return.
 </div>
 
+Patch the [=navigate=] algorithm to allow delaying or discarding of a cross-origin navigations in a [=prerendering browsing context=] as follows:
+
+<div algorithm="navigate delay cross-origin patch">
+  In [=navigate=], append the following steps after the cross-origin redirect check (currently step 13.1):
+
+  1. Let |crossOrigin| be <var ignore>hasCrossOriginRedirects</var>.
+
+  1. f |bc|'s <a spec="HTML">container</a> is not null, and |bc|'s <a spec="HTML">container</a>'s [=Node/node document=]'s	[=Document/origin=] is not [=same origin=] as <var ignore>currentURL</var>'s [=url/origin=], then set |crossOrigin| to true.
+
+  1. If |crossOrigin| is true and |browsingContext|'s [=top-level browsing context=] is a [=prerendering browsing context=], then wait until |browsingContext| is [=prerendering browsing context/activate|activated=].
+
+  User agents may choose to [=discard=] a [=prerendering browsing context=] when its document or one of its iframes navigates to a cross-origin URL.
```

Comment:
This is basically an alternative to waiting, right? If so, I'd make it explicit:

1. If crossOrigin is true and browsingContext's TLBC is a prerendering BC, then the user agent must either wait to continue this algorithm until browsingContext is activated, or discard |browsingContext|'s TLBC.